nifi-nar-bundles/nifi-solr-bundle/nifi-solr-processors/src/main/java/org/apache/nifi/processors/solr/FetchSolr.java

```diff
@@ -0,0 +1,401 @@
+                    ) {
+                        addFacetsFromSolrResponseToJsonWriter(response, writer);
+                    }
+                });
+                flowFileFacets = session.putAttribute(flowFileFacets, CoreAttributes.MIME_TYPE.key(), MIME_TYPE_JSON);
+                flowFileFacets = session.putAllAttributes(flowFileFacets, attributes);
+                session.transfer(flowFileFacets, FACETS);
+            }
+
+            if (searchComponents.contains(StatsParams.STATS)) {
+                FlowFile flowFileStats = session.create(flowFileOriginal);
+                flowFileStats = session.write(flowFileStats, out -> {
+                    try (
+                            final OutputStreamWriter osw = new OutputStreamWriter(out);
+                            final JsonWriter writer = new JsonWriter(osw)
+                    ) {
+                        addStatsFromSolrResponseToJsonWriter(response, writer);
+                    }
+                });
+                flowFileStats = session.putAttribute(flowFileStats, CoreAttributes.MIME_TYPE.key(), MIME_TYPE_JSON);
+                flowFileStats = session.putAllAttributes(flowFileStats, attributes);
+                session.transfer(flowFileStats, STATS);
+            }
+
+        } catch (Exception e) {
+            flowFileOriginal = session.penalize(flowFileOriginal);
+            flowFileOriginal = session.putAttribute(flowFileOriginal, EXCEPTION, e.getClass().getName());
+            flowFileOriginal = session.putAttribute(flowFileOriginal, EXCEPTION_MESSAGE, e.getMessage());
+            session.transfer(flowFileOriginal, FAILURE);
+            logger.error("Failed to execute query {} due to {}. FlowFile will be routed to relationship failure", new Object[]{solrQuery.toString(), e}, e);
+        }
+
+        if (!flowFileOriginal.isPenalized())
+            session.transfer(flowFileOriginal, ORIGINAL);
+    }
+
+    private Map<String,String[]> parseSolrParams(final ProcessContext context, final FlowFile flowFile) {
+        final MultiMapSolrParams newSolrParams = SolrRequestParsers.parseQueryString(context.getProperty(SOLR_QUERY_STRING).evaluateAttributeExpressions(flowFile).getValue());
+        return Collections.unmodifiableMap(newSolrParams.getMap());
+    }
+
+    private Set<String> extractSearchComponents(Map<String,String[]> solrParams) {
+        final Set<String> searchComponentsTemp = new HashSet<>();
+        for (final String searchComponent : SUPPORTED_SEARCH_COMPONENTS)
+            if (solrParams.keySet().contains(searchComponent))
+                if (SEARCH_COMPONENTS_ON.contains(solrParams.get(searchComponent)[0]))
+                    searchComponentsTemp.add(searchComponent);
+        return Collections.unmodifiableSet(searchComponentsTemp);
+    }
+
+    private static void addStatsFromSolrResponseToJsonWriter(final QueryResponse response, final JsonWriter writer) throws IOException {

```

Comment:
It might be simpler to tie this to adding wt=json and just pull out the entire stats/facets branch and put that into the flowfile.

Reply:
See [this](https://stackoverflow.com/a/11625103/284538) for an example if you haven't done it before.

Reply:
The parameter wt only works for communication with Solr via http. This processor uses Solrj (which is highly recommended for SolrCloud mode). Solrj is only capable to handle binary response and XML. However, the processor makes use of the records functions. Therefore, the user easily can define an custom format (JSON, CSV, ...).

Reply:
Ok. I could have sworn that I used wt=json that way, but it's been a while.

Reply:
In the [StackOverflow issue](https://stackoverflow.com/a/11625103/284538) you mentioned above this is also mentioned. 
For more detailed information see here: https://lucene.apache.org/solr/guide/6_6/using-solrj.html